ee/clickhouse/queries/retention/retention_actors.py

```diff
@@ -0,0 +1,202 @@
+
+        target_params.update(
+            {
+                "target_start_date": reference_date_from.strftime(
+                    "%Y-%m-%d{}".format(" %H:%M:%S" if self._filter.period == "Hour" else " 00:00:00")
+                ),
+                "target_end_date": reference_date_to.strftime(
+                    "%Y-%m-%d{}".format(" %H:%M:%S" if self._filter.period == "Hour" else " 00:00:00")
+                ),
+            }
+        )
+
+        return (
+            RETENTION_PEOPLE_SQL.format(
+                target_event_query=target_event_query,
+                returning_query=return_query_formatted,
+                filters=prop_filters,
+                actor_field_name=actor_field_name,
+                person_join=""
+                if self.is_aggregating_by_groups
+                else f"JOIN ({GET_TEAM_PERSON_DISTINCT_IDS}) pdi on e.distinct_id = pdi.distinct_id",
+            ),
+            {
+                "team_id": self._team.pk,
+                "start_date": date_from.strftime(
+                    "%Y-%m-%d{}".format(" %H:%M:%S" if self._filter.period == "Hour" else " 00:00:00")
+                ),
+                "end_date": date_to.strftime(
+                    "%Y-%m-%d{}".format(" %H:%M:%S" if self._filter.period == "Hour" else " 00:00:00")
+                ),
+                "offset": self._filter.offset,
+                **target_params,
+                **return_params,
+                **prop_filter_params,
+            },
+        )
+
+
+class ClickhouseRetentionActorsByPeriod(ActorBaseQuery):
+    DISTINCT_ID_TABLE_ALIAS = "pdi"
+    EVENT_TABLE_ALIAS = "e"
+    _filter: RetentionFilter
+
+    def __init__(self, team: Team, filter: RetentionFilter):
+        super().__init__(team, filter)
+
+    @cached_property
+    def is_aggregating_by_groups(self) -> bool:
+        return self._filter.aggregation_group_type_index is not None
+
+    def actor_query(self) -> Tuple[str, Dict]:
+        period = self._filter.period
+        is_first_time_retention = self._filter.retention_type == RETENTION_FIRST_TIME
+        trunc_func = get_trunc_func_ch(period)
+        prop_filters, prop_filter_params = parse_prop_clauses(self._filter.properties)
+
+        target_query, target_params = _get_condition(self._filter.target_entity, table="e")
+        target_query_formatted = f"AND {target_query}"
+        return_query, return_params = _get_condition(self._filter.returning_entity, table="e", prepend="returning")
+        return_query_formatted = f"AND {return_query}"
+
+        actor_field_name = f"{get_aggregation_target_field(self._filter.aggregation_group_type_index, self.EVENT_TABLE_ALIAS, self.DISTINCT_ID_TABLE_ALIAS)} AS actor_id"
+        person_join = (
+            ""
+            if self.is_aggregating_by_groups
+            else f"JOIN ({GET_TEAM_PERSON_DISTINCT_IDS}) pdi on e.distinct_id = pdi.distinct_id"
```

Comment:
Please update this with the new method: ee/clickhouse/queries/person_distinct_id_query.py